dns_discover.go

```diff
@@ -40,6 +41,20 @@ func discoverDNS(domain string, port int) (servers []string, ttl time.Duration,
 	return
 }
 
+// retryingFindTXT will, on any DNS failure, retry for up to 10 minutes before
+// giving up and returning an empty []string of records
+func retryingFindTXT(fqdn string) (records []string, ttl time.Duration, err error) {
+	err = backoff.Retry(
+		func() error {
+			records, ttl, err = findTXT(fqdn)
+			if err != nil {
+				log.Error("Retrying DNS query. Query failed with: %s", err.Error())
+			}
+			return err
+		}, backoff.NewExponentialBackOff())
+	return
```

Comment:
Any reason for using a naked `return` here rather than `return records, ttl, err`? IMO it seems like the latter would be easier for a reader to understand.